semgrep-core/src/matching/Generic_vs_generic.ml

```diff
@@ -374,6 +374,20 @@ let rec m_name a b =
       m_name a (B.Id (idb, B.empty_id_info ()))
       >||> (* try this time a match with the resolved entity *)
       m_name a (H.name_of_ids dotted)
+      >||>
```

Comment:
This may introduce a slowdown when we do this in m_name. Maybe we should do it only when we know we are in a type context where subtyping can be used (new Foo, raise Bar, class X extends Bla, etc.).